### PR TITLE
Support identifying btf type for btf-based tracing

### PR DESCRIPTION
**Pull request by chengshuyi**
k(ret)func and iter are btf-based tracing, which can detect whether dereferencing is safe through btf, so dereferencing does not need to call the helper function bpf_probe_read. This patch mainly modifies the semantic and codegen modules

semantic:
- Handle the ctx of iter, mark is_btftype as true
- When dealing with ArrayAccess, Unop, FieldAccess, inherit is_btftype attribute

codegen:
- If it is a string type, use CREATE_MEMCPY directly to load the data
- For other types, load data through CreateDatastructElemLoad instead of CreateProbeRead

##### Checklist

- [ ] Language changes are updated in `man/adoc/bpftrace.adoc` and if needed in `docs/reference_guide.md`
- [ ] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [x] The new behaviour is covered by tests

### CODE REVIEW

File: src/ast/passes/codegen_llvm.cpp [hunk 1]
```diff
@@ -3561,6 +3570,14 @@ void CodegenLLVM::probereadDatastructElem(Value *src_data,
...
       }
     }
+    else if (data_type.is_btftype)
+    {
+      expr_ = b_.CreateDatastructElemLoad(
+          elem_type,
+          b_.CreateIntToPtr(src, dst_type->getPointerTo()),
+          true,
+          data_type.GetAS());
+    }
```
Comment (viktormalik) [inline src/ast/passes/codegen_llvm.cpp:3580]: This is exactly the same call as on lines 3545-3549, can we merge them?

Comment (chengshuyi) [reply src/ast/passes/codegen_llvm.cpp:3580]: Ok.

**Comment by chengshuyi** [comment]
> E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/j/javascript-common/javascript-common_11_all.deb  Could not connect to archive.ubuntu.com:80 (91.189.91.38), connection timed out Could not connect to archive.ubuntu.com:80 (185.125.190.36), connection timed out Could not connect to archive.ubuntu.com:80 (91.189.91.39), connection timed out Could not connect to archive.ubuntu.com:80 (185.125.190.39), connection timed out [IP: 185.125.190.39 80]

It seems ubuntu package server is not stable which causes ci fail.

**Comment by viktormalik** [comment]
I'm surprised that this didn't break any codegen test. It looks like we don't have a test for kfunc+dereference. Maybe this is a good time to add one?

**Comment by chengshuyi** [comment]
> I'm surprised that this didn't break any codegen test. It looks like we don't have a test for kfunc+dereference. Maybe this is a good time to add one?

Ok, will do

**Review by danobi — APPROVED** [review]
great patch! lgtm

**Review by viktormalik — APPROVED** [review]
Looks good, thanks!

**Pull request merged**